mesh/mesh.go

```diff
@@ -236,6 +236,22 @@ func (msh *Mesh) GetLayer(i types.LayerID) (*types.Layer, error) {
 	return l, nil
 }
 
+// GetLayerHash returns layer hash for received blocks
+func (msh *Mesh) GetLayerHash(layerID types.LayerID) types.Hash32 {
+	h, err := msh.recoverLayerHash(layerID)
+	if err == nil {
+		return h
+	}
+	if err == database.ErrNotFound {
+		// layer hash not persisted. i.e. contextual validity not yet determined
+		lyr, err := msh.GetLayer(layerID)
```

Comment:
this means that layer hash might be changed when layer validity is determined, is this wanted behaviour?